data/transactions/logic/debugger.go

```diff
@@ -192,10 +200,32 @@ func valueDeltaToValueDelta(vd *basics.ValueDelta) basics.ValueDelta {
 	}
 }
 
+// parseCallStack initializes an array of CallFrame objects from the raw
+// callstack.
+func (d *DebugState) parseCallstack(callstack []int) []CallFrame {
+	callFrames := make([]CallFrame, 0)
+	lines := strings.Split(d.Disassembly, "\n")
+	for _, pc := range callstack {
+		// The callsub is pc - 3 from the callstack pc
+		callsubLineNum := d.PCToLine(pc - 3)
+		callSubLine := strings.Fields(lines[callsubLineNum])
+		label := ""
+		if callSubLine[0] == "callsub" {
```

Comment:
@algochoi As I understand it, to reach the highlighted line, a program must successfully assemble.  So, _in principle_, the highlighted parsing always succeeds.  

Nevertheless, I'm wondering:  Is there a way to avoid the error case or otherwise handle the error?

I performed some review before asking though I suspect I don't fully understand the space I'm asking about.  Part of the reason for asking is to further my understanding.

From what I saw, it seems avoiding the error case isn't possible without serious effort.  If true, I'm wondering if there's conventions for signaling a failed invariant to the end user.

Reply:
> So, in principle, the highlighted parsing always succeeds.

I think so too - it should always succeed unless we change the `callsub` semantics somehow. Currently `callsub` just pushes `pc+3` to the stack so I just hardcoded `pc-3` to mean the `callsub` line. If this change and this line isn't updated things might get scary afterwards, but I'm guessing existing opcode changes would be almost never done or done very carefully. 

> Nevertheless, I'm wondering: Is there a way to avoid the error case or otherwise handle the error?

If the current line is not a `callsub` then I just leave the `label` blank. I don't think it's a huge deal because the only impact is that the label name won't show in the Call stack section of CDT. Labels/Call frames didn't show names before this PR anyway, so I don't think it violates anything backwards. I added the if statement just in case because I didn't want to access `callSubLine[1]` until I was sure I was on a line that had at least two "words" or pc on it.

Reply:
@algochoi Thanks for the discussion here. I think it addresses my question and I'm good to resolve.

Reply:
I added a `TestParseCallstack` so that test will fail and notify us if the `callsub` semantics have changed - hopefully some peace of mind for both of us.